cmd/gitserver/internal/git/gitcli/mergebase.go

```
@@ -47,3 +47,44 @@ func (g *gitCLIBackend) MergeBase(ctx context.Context, baseRevspec, headRevspec
 
 	return api.CommitID(bytes.TrimSpace(stdout)), nil
 }
+
+func (g *gitCLIBackend) OctopusMergeBase(ctx context.Context, revspecs ...string) (api.CommitID, error) {
+	if len(revspecs) < 2 {
+		return "", errors.New("at least two revspecs must be given")
+	}
+
+	args := make([]string, 0, len(revspecs)+3)
+	args = append(args, "merge-base", "--octopus", "--")
+	args = append(args, revspecs...)
+
+	out, err := g.NewCommand(
+		ctx,
+		WithArguments(args...),
+	)
+	if err != nil {
+		return "", err
+	}
+
+	defer out.Close()
+
+	stdout, err := io.ReadAll(out)
+	if err != nil {
+		// Exit code 1 and empty output most likely means that no common merge-base was found.
+		var e *commandFailedError
+		if errors.As(err, &e) {
+			if e.ExitStatus == 1 {
+				if len(e.Stderr) == 0 {
+					return "", nil
+				}
```

When can this trigger? I expect `api.CommitID` values to never be `""`, so this seems quite counter-intuitive. It would be better to return a sentinel error `NoMergeBaseFound`  and document that in the function's doc comment, so that one doesn't accidentally keep chugging along with an empty CommitID.